apps/site/assets/ts/schedule/components/schedule-finder/TableRow.tsx

```diff
@@ -1,179 +1,138 @@
-import React, { useState, ReactElement } from "react";
-import { ScheduleWithFare, ScheduleInfo } from "../__schedule";
+import React, { useEffect, useState, useReducer, ReactElement } from "react";
+import { reducer } from "../../../helpers/fetch";
+import { Journey, TripInfo } from "../__trips";
 import { RoutePillSmall } from "./UpcomingDepartures";
 import { modeIcon, caret } from "../../../helpers/icon";
 import { handleReactEnterKeyPress } from "../../../helpers/keyboard-events";
 import { breakTextAtSlash } from "../../../helpers/text";
+import { TripDetails } from "./TripDetails";
+import { UserInput } from "../ScheduleFinder";
 
-const totalMinutes = (schedules: ScheduleInfo): string => schedules.duration;
-
-interface Props {
+interface TableRowProps {
+  input: UserInput;
+  journey: Journey;
   isSchoolTrip: boolean;
-  schedules: ScheduleInfo;
   anySchoolTrips: boolean;
 }
 
-const TripInfo = ({
-  schedules
-}: {
-  schedules: ScheduleInfo;
-}): ReactElement<HTMLElement> => {
-  const lastTrip = schedules.schedules[schedules.schedules.length - 1];
-  return (
-    <tr>
-      <td colSpan={3}>
-        <div className="schedule-table__subtable-trip-info">
-          <div className="schedule-table__subtable-trip-info-title u-small-caps u-bold">
-            Trip length
-          </div>
-          {schedules.schedules.length} stops, {totalMinutes(schedules)} minutes
-          total
-        </div>
-        <div className="schedule-table__subtable-trip-info">
-          <div className="schedule-table__subtable-trip-info-title u-small-caps u-bold">
-            Fare
-          </div>
-          {lastTrip.price}
-          <a
-            className="schedule-table__subtable-trip-info-link"
-            href={lastTrip.fare_link}
-          >
-            View fares
-          </a>
-        </div>
-      </td>
-    </tr>
-  );
-};
+interface AccordionProps {
+  input: UserInput;
+  journey: Journey;
+  contentCallback: () => ReactElement<HTMLElement>;
```

Comment:
I think naming this a "callback" is potentially confusing — it's a component, really. `contentComponent`?